### PR TITLE
IA-3463 improve payment lot creation

### PR DESCRIPTION
**Pull request by quang-le**
Explain what problem this PR is resolving

Related JIRA tickets : IA-3463

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [X] My migrations file are included
- [X] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc
In the code

## Changes

- Move all code related to payment lot creation to new task "create_payment_lot"
- Make sure payment lot is deleted if there's any error with the conversion of any potential payment into a payment
- Add "task" field to potential payment
- Add migration
- Add custom message when creating payment lot
- Add and update tests

## How to test

- Make sure to have at least 1 potential  payment, i.e.: have accepted org unit change requests
- Select one or several potential payments and create a payment lot

Then:
- Selected potential payments should have disappeared from table
- In task screen: a successful task should be visible
- In payment lots screen: the new payment lot should be visible

## Print screen / video

https://github.com/user-attachments/assets/6988e808-8bc5-417e-8ef7-9aa3028ddb4b

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/app/translations/fr.json [hunk 1]
```diff
@@ -1228,6 +1228,7 @@
     "iaso.snackBar.newEmptyVersionSavedSuccess": "Nouvelle version vide sauvegardée avec succès",
     "iaso.snackBar.patchTaskError": "Une erreur est survenue lors de la mise à jour de la tâche.",
     "iaso.snackBar.patchTaskSuccess": "La tâche a été mise à jour.",
+    "iaso.snackBar.paymentLotTaskLaunched": "La création du lot de paiments peut prendre un instant. Essayez de rafraîchir la page dans quelques secondes",
```
Comment (mathvdh) [inline hat/assets/js/apps/Iaso/domains/app/translations/fr.json]: lot de pai**e**ments ?

File: hat/assets/js/apps/Iaso/domains/app/translations/fr.json [hunk 2]
```diff
@@ -320,8 +320,8 @@
     "iaso.groups.sourceVersion": "Version de la source",
     "iaso.groups.update": "Mettre le groupe à jour",
     "iaso.groupsets.dialog.delete": "Etes-vous certain de vouloir effacer cet ensemble de groupe?",
-    "iaso.groupsets.dialog.deleteText":"Cette opération ne peut-être annulée",
-    "iaso.groupsets.groupBelonging": "Appartenance aux groupes", 
+    "iaso.groupsets.dialog.deleteText": "Cette opération ne peut-être annulée",
```
Comment (mathvdh) [inline hat/assets/js/apps/Iaso/domains/app/translations/fr.json]: Why peut-etre and not peut etre

**Review by mathvdh — CHANGES_REQUESTED** [review]
Working good locally, a few minor comments : 

The message : 
`The creation of the payment lot may take a few moments. Try refreshing the page in a few seconds`

Is confusing because even if you refresh after its created, nothing happens. You need to navigate to the other page ( From potential payments to payment lots) Maybe we should explain it or add a link to the other page ?

**Comment by mathvdh** [comment]
Another question I have is that in the ticket you mention you are gonna move the task field from one place to another, but here you just add it on potentialpayment but not remove it from the other model

**Comment by quang-le** [comment]
> Another question I have is that in the ticket you mention you are gonna move the task field from one place to another, but here you just add it on potentialpayment but not remove it from the other model

True, I half-forgot, I half didn't dare for fear of unknowingly break something

**Comment by quang-le** [comment]
> Another question I have is that in the ticket you mention you are gonna move the task field from one place to another, but here you just add it on potentialpayment but not remove it from the other model

Actually, I realized I need it to filter out PaymentLots that are in the process of being created from the API response for the Payment Lots table

**Review by mathvdh — APPROVED** [review]
Go go go

**Pull request merged**